src/Microsoft.AspNetCore.Mvc.Razor.Precompilation.Tools/Internal/PrecompileDispatchCommand.cs

```diff
@@ -0,0 +1,164 @@
+using Microsoft.AspNetCore.Mvc.Razor.Precompilation.Design.Internal;
+using Microsoft.DotNet.InternalAbstractions;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.Extensions.CommandLineUtils;
+using Microsoft.Extensions.Internal;
+using NuGet.Frameworks;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Precompilation.Internal
+{
+    public class PrecompileDispatchCommand
+    {
+        private CommonOptions Options { get; } = new CommonOptions();
+
+        private CommandOption FrameworkOption { get; set; }
+
+        private CommandOption ConfigurationOption { get; set; }
+
+        private CommandOption OutputPathOption { get; set; }
+
+        private NuGetFramework TargetFramework { get; set; }
+
+        private CommandOption BuildBasePathOption { get; set; }
+
+        private string ProjectPath { get; set; }
+
+        private string Configuration { get; set; }
+
+        public void Configure(CommandLineApplication app)
+        {
+            Options.Configure(app);
+            FrameworkOption = app.Option(
+                "-f|--framework",
+                "Target Framework",
+                CommandOptionType.SingleValue);
+
+            ConfigurationOption = app.Option(
+                "-c|--configuration",
+                "Configuration",
+                CommandOptionType.SingleValue);
+
+            OutputPathOption = app.Option(
+                "-o|--output-path",
+                "Published path of the application.",
+                CommandOptionType.SingleValue);
+
+            app.OnExecute(() => Execute());
+        }
+
+        private int Execute()
+        {
+            ProjectPath = GetProjectPath();
+            Configuration = ConfigurationOption.Value() ?? DotNet.Cli.Utils.Constants.DefaultConfiguration;
+            TargetFramework = GetTargetFramework();
+
+            var outputPaths = GetOutputPaths();
+            var applicationName = Path.GetFileNameWithoutExtension(outputPaths.CompilationFiles.Assembly);
+            var dispatchArgs = new List<string>
+            {
+                "--debug",
+                ProjectPath,
+                "--application-name",
+                applicationName,
+                "--output-path",
+                GetOutputPath(),
+                "--content-root",
+                Options.ContentRootOption.Value() ?? Directory.GetCurrentDirectory(),
+            };
+
+            if (Options.ConfigureCompilationType.HasValue())
+            {
+                dispatchArgs.Add("--configure-compilation-type");
+                dispatchArgs.Add(Options.ConfigureCompilationType.Value());
+            }
+
+            var toolName = typeof(Design.Program).GetTypeInfo().Assembly.GetName().Name;
+            var dispatchCommand = DotnetToolDispatcher.CreateDispatchCommand(
+                dispatchArgs,
+                TargetFramework,
+                Configuration,
+                outputPath: outputPaths.RuntimeOutputPath,
+                buildBasePath: null,
```

Comment:
We calculate the output path which is where publish would have dropped the output to. We don't need the build base path in this case.